lib/iris/tests/unit/io/test_expand_filespecs.py

```diff
@@ -4,48 +4,42 @@
 # See LICENSE in the root of the repository for full licensing details.
 """Unit tests for the `iris.io.expand_filespecs` function."""
 
-# Import iris.tests first so that some things can be initialised before
-# importing anything else.
-import iris.tests as tests  # isort:skip
-
 import os
 from pathlib import Path
-import shutil
-import tempfile
+import re
 import textwrap
 
+import pytest
+
 import iris.io as iio
 
 
-class TestExpandFilespecs(tests.IrisTest):
-    def setUp(self):
-        tests.IrisTest.setUp(self)
-        self.tmpdir = os.path.realpath(tempfile.mkdtemp())
+class TestExpandFilespecs:
+    @pytest.fixture(autouse=True)
+    def _setup(self, tmp_path):
+        self.tmpdir = tmp_path
         self.fnames = ["a.foo", "b.txt"]
         for fname in self.fnames:
-            with open(os.path.join(self.tmpdir, fname), "w") as fh:
```

Comment:
You've removed `os.path` here, but it's still in use through the rest of the file. It can all be replaced with `self.tmp_dir / foo`.

Reply:
In fact you can even use `(self.tmp_dir / foo).open(...`

Reply:
That's a bit of refactoring that causes problems in some places. A lot of converting to `str` if I use `self.tmpdir/foo`.

Want me to action it? I don't mind!